tests/test_complete_rules.py

```diff
@@ -305,3 +343,21 @@ def test_complete_rules(rule):
     assert session.run(textwrap.dedent(rule.INPUT_SOURCE)) == textwrap.dedent(
         rule.EXPECTED_SOURCE
     )
+
+
+@pytest.mark.parametrize(
+    "rule",
+    [
+        OnlyKeywordArgumentDefaultNotSetCheckRule,
+    ]
+)
+def test_check_rules(rule):
+    source = textwrap.dedent(rule.INPUT_SOURCE)
+    tree = ast.parse(source)
+
+    session = Session([rule])
+    rule = session._initialize_rules(tree, source, None).pop()
+
+    for node in ast.walk(tree):
+        with suppress(AssertionError):
+            assert rule.match(node) is True
```

Comment:
And once you change the test to use the transformation format, you can remove this test and use the test above to parameterize your rule. It would automatically work.